plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/CitrixResourceBase.java

```diff
@@ -1402,6 +1402,12 @@ public VM createWorkingVM(final Connection conn, final String vmName, final Stri
             vbdr.userdevice = Long.toString(volumeTO.getDeviceId());
             vbdr.mode = Types.VbdMode.RW;
             vbdr.type = Types.VbdType.DISK;
+            Long deviceId = volumeTO.getDeviceId();
+            if (deviceId != null) {
+                if (!isDeviceUsed(conn, vm, deviceId) || deviceId.longValue() > 3) {
```

Comment:
Calling ``longValue`` is unnecessary due to autoboxing.  Also, ``3`` seems like a magic value.  What is its significance?